File: linear_programming/python/lp.py
from pulp import LpMaximize, LpProblem, LpVariable

# Create a linear programming problem
problem = LpProblem("Maximize_Profit", LpMaximize)

# Define decision variables
product_a = LpVariable("Product_A", lowBound=0, cat="Integer")
product_b = LpVariable("Product_B", lowBound=0, cat="Integer")

# Define the objective function (profit to maximize)
profit_per_unit_a = 10
profit_per_unit_b = 15
problem += (profit_per_unit_a * product_a) + (profit_per_unit_b * product_b)

# Define resource constraints
resource_1_usage_a = 2
resource_1_usage_b = 3
resource_1_avail = 100
resource_2_usage_a = 4
resource_2_usage_b = 2
resource_2_avail = 80

problem += (resource_1_usage_a * product_a) + (resource_1_usage_b * product_b) <= resource_1_avail
problem += (resource_2_usage_a * product_a) + (resource_2_usage_b * product_b) <= resource_2_avail

# Define production capacity constraints
max_units_product_a = 20
max_units_product_b = 30

problem += product_a <= max_units_product_a
problem += product_b <= max_units_product_b

# Solve the problem
problem.solve()

# Print the results
print(f"Product A units to produce: {product_a.varValue}")
print(f"Product B units to produce: {product_b.varValue}")
print(f"Total Profit: ${problem.objective.value()}")

File: linear_programming/julia/lp.jl
using JuMP
using GLPK

# Create a JuMP model with GLPK solver
model = Model(optimizer_with_attributes(GLPK.Optimizer))

# Define the decision variables
@variable(model, product_a >= 0, Int)
@variable(model, product_b >= 0, Int)

# Set the objective function to maximize profit
@objective(model, Max, 10 * product_a + 15 * product_b)

# Add resource constraints
@constraint(model, 2 * product_a + 3 * product_b <= 100)
@constraint(model, 4 * product_a + 2 * product_b <= 80)

# Add production capacity constraints
@constraint(model, product_a <= 20)
@constraint(model, product_b <= 30)

# Solve the linear programming problem
optimize!(model)

# Get the optimal values of the decision variables
opt_product_a = value(product_a)
opt_product_b = value(product_b)

# Calculate the total profit
total_profit = objective_value(model)

# Print the results
println("Product A units to produce: ", opt_product_a)
println("Product B units to produce: ", opt_product_b)
println("Total Profit: \$", total_profit)
